### DISCUSSION TITLE
Moved Photoprism Hard Drive from Windows to Linux, How to Fix Permission and Ownership

### DISCUSSION THREAD

**Discussion by the3dsandwich** [photoprism/photoprism#1364] [Q&A] [answered]
Hi,

I am in the process of switching my whole photo library to Photoprism, and I plan to use a spare Linux (Ubuntu) PC to act as the server. However, before I did that, I used my main PC (Windows OS with Docker) to handle import and indexing since it is much faster.

Everything needed for Photoprism is in my external hard drive (docker-compose.yml file, originals, database, storage, rendered sidecar files, etc), but after digging into it I think it's the problem with permissions. more specifically, file ownership settings.

The hard drive is in exFAT format, I can change mounting options (which fixes file rw permissions) but ownership I just cannot get it to work.

What ownership should I change for which files/directories?

Thanks

**TL;DR (realized I didn't line out my problem clear enough): Imported/indexed photos using Windows, moved hard drive to Linux to serve, now I cannot docker-compose up because file ownership problems.**

**Comment by the3dsandwich** [comment]
My file structure looks like this:

/media/Photoprism/
- originals/: imported library of my photos
- photoprism/docker-compose.yml
- photoprism/database: automatically generated database directory
- photoprism/storage: automatically generated storage directory (sidecar, etc)

The error by running `sudo docker-compose up` goes along the lines of `chown: changing ownership of './database/<files>': Operation not permitted`

**Comment by G2G2G2G** [comment, marked as answer]
I'd move the files somewhere else and change your drive to ext4 100% and move back.
exfat does not support user permissions, you can mount the drive in etc/fstab with options like `defaults,uid=xxxx,gid=xxxx` or such you might be able to chmod 777 it all but prob not. haven't bothered with exfat in several years.

**Comment by the3dsandwich** [comment]
So say if I did make two transfers to switch them to ext4, what permissions should I give or change? Will PhotoPrism handle it for me when I re-run `docker-compose up` or should I chmod and chown it manually?

Thanks for the reply!

**Comment by G2G2G2G** [comment]
Yea if you copy it identically, and give it 777 permissions (for now, so it works) and make the owner /group of the photoprism it should work fine.
